Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -48,10 +48,50 @@ namespace Metal {
 #define DECLARE_FORWARD_PROGRESS "volatile uint32_t __wgslEnsureForwardProgress = 0; if (!__wgslEnsureForwardProgress)"
 #define CHECK_FORWARD_PROGRESS "if (++__wgslEnsureForwardProgress == 4294967295u) break;"
 
+#define STRINGIFY_(__x) #__x##_s
+#define STRINGIFY(__x) STRINGIFY_(__x)
+
+#define DEFINE_HELPER(__name, ...) \
+    void emit##__name() { m_output.append(STRINGIFY(__VA_ARGS__)); } \
+    bool didEmit##__name { false };
+
+#define DEFINE_TRIG_HELPER(__name, __capitalizedName, ...) \
+    DEFINE_HELPER(__capitalizedName,  \
+    template <typename T> \
+    T __wgslAsin(T value) \
```

Comment:
does `__wgslAsin` need to be `STRINGIFY(__wgsl, __capitalizedName)` or am I misreading the macro?

Reply:
indeed this is observed in https://webgpu.github.io/webgpu-samples/?sample=particles